.github/workflows/sync-branches.yml

```diff
@@ -0,0 +1,26 @@
+name: Sync branches
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  sync:
+    name: Sync branches
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Synchronize source to target branch
+        run: |
+          git checkout -f "${SOURCE_BRANCH}"
```

Comment:
I somehow feel nervous to use env variables, even though they were already used in Jenkins.

Reply:
That's fine.

Can you confirm whether this step worked in GH action with the [current default read permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token)?

Reply:
AFAIS, we use the default permissive mode, which grants read/write permissions for all scopes.
I tested the workflow using a different source and target branch.
https://github.com/elastic/apm-agent-java/actions/runs/4502205643/jobs/7923696347

Reply:
Cool, I'd say to put the explicit permissions just in case those are changed in the future.